assertions/src/main/java/io/camunda/zeebe/process/test/assertions/ProcessInstanceAssert.java

```diff
@@ -634,7 +634,9 @@ public ProcessInstanceAssert hasNotCalledProcess() {
     final boolean hasCalledProcess = getCalledProcessRecords().stream().findAny().isPresent();
 
     assertThat(hasCalledProcess)
-        .withFailMessage("A process was called from this process")
+        .withFailMessage("A process was called from this process, distinct called processes are: %s",
+            getCalledProcessRecords().stream().map(
+                x -> x.getValue().getBpmnProcessId()).distinct().collect(Collectors.toList()).toString())
```

Comment:
🔧 One more minor change before I can merge it. We already get the caller process records on line 634. Please store the result of this in a variable instead of getting them again here. E.g.:

```java
  public ProcessInstanceAssert hasNotCalledProcess() {
    final Set<String> calledProcesses =
        getCalledProcessRecords().stream()
            .map(process -> process.getValue().getBpmnProcessId())
            .collect(Collectors.toSet());

    assertThat(!calledProcesses.isEmpty())
        .withFailMessage(
            "A process was called from this process, distinct called processes are: %s",
            calledProcesses)
        .isFalse();
    return this;
  }
```

You will have to swap the processes in the failure message of your unit test.

Reply:
Ah, Collectors.toSet() removes the need for distinct. I still gotta get more used to streams in Java. 
Thanks for the reviewing!